CMakeLists.txt

```diff
@@ -98,6 +112,22 @@ endif()
 
 find_package(PkgConfig REQUIRED)
 
+############# Set AWS CRT based variables #############
+if(BUILD_AWS_CRT)
+    if(APPLE)
+        set(CFLIBRARY "-framework CoreFoundation")
```

Comment:
Since this is only set for Apple could we unset it right before this just in case it accidentally gets set somehow so later when we call target link we can't accidentally have something which is non-empty for non-Apple., not sure if that is a real concern.

Reply:
Its not a problem. Tested it out. But, set it to empty anyways in the new commit.